base/abstractarray.jl

```diff
@@ -1962,6 +1981,258 @@ function typed_hvcat(::Type{T}, rows::Tuple{Vararg{Int}}, as...) where T
     T[rs...;]
 end
 
+# nd concatenation
+
+hvncat(dims::Tuple{Vararg{Int}}, row_first::Bool) = []
+
+function hvncat(dims::Tuple{Vararg{Int, N}}, row_first::Bool, xs...) where N
+    N == 1 && return vcat(xs...)
+    row_first && N == 2 && return hvcat(ntuple(x->dims[2], length(xs) ÷ dims[2]), xs...)
```

Comment:
Just repeating this here so it doesn't get lost: I think it would be nicer to introduce another layer of dispatch with `_hvncat` here instead of having to copy these checks everywhere.

Reply:
Got it done!